### DISCUSSION TITLE
Adonis V5 socket.io

### DISCUSSION THREAD

**Discussion by UhoMazino** [adonisjs/core#1959] [Help]
Hi, im trying  add socket.io to my adonis project using this guide: https://preview.adonisjs.com/blog/socketio-with-adonisjs-v5
```js
//~/app/Services/Ws.ts
import socketIo from 'socket.io'
import Server from '@ioc:Adonis/Core/Server'

class Ws {
  public isReady = false
  public io: socketIo.Server

  public start(callback: (io: socketIo.Server, socket: socketIo.Socket) => void) {
    this.io = socketIo(Server.instance!)
    this.io.on('connection', socket => callback(this.io, socket))
    this.isReady = true
  }
}
export default new Ws()

//~/start/socket.ts

import Ws from 'App/Services/Ws'

Ws.start((io, socket) => {
  console.log(io, socket)
  socket.emit('news', 'hello world')
})
```
But when i connect from client, i have 404 error
```
code: "E_ROUTE_NOT_FOUND"
message: "E_ROUTE_NOT_FOUND: Cannot GET:/socket.io/"
```
I need specific route and handler or i do wrong?

**Comment by thetutlage** [comment]
Is `start/socket` file being loaded on the application boot. Just write a console.log statement inside it to verify

**Reply by UhoMazino** [reply]
> Is `start/socket` file being loaded on the application boot. Just write a console.log statement inside it to verify
> Did you add your `socket.ts` file to the array `preloads` of your `.adonisrc.json` file ?

Yap, its started.
![image](https://user-images.githubusercontent.com/32494642/100356726-ad464c80-3004-11eb-9913-c3a0f581b96c.png)
![image](https://user-images.githubusercontent.com/32494642/100356865-debf1800-3004-11eb-9431-3d9011aa67dc.png)

**Comment by DraftProducts** [comment]
Did you add your `socket.ts` file to the array `preloads` of your `.adonisrc.json` file ?

**Comment by choirool** [comment]
This work for me

```
//~/app/Services/Ws.ts

import { Server as socketServer } from 'socket.io'
import Server from '@ioc:Adonis/Core/Server'

class Ws {
  public isReady = false
  public io: socketServer

  public start(callback: (socket) => void) {
    this.io = new socketServer(Server.instance!, {
      cors: {
        origin: 'http://127.0.0.1:3333',
        methods: ['GET', 'POST'],
        allowedHeaders: ['*'],
        credentials: true,
      },
    })

    this.io.on('connection', callback)
    this.isReady = true
  }
}

/**
 * This makes our service a singleton
 */
export default new Ws()

```
